scalableminds/webknossos
# Make mappings work with 8 and 16 bit segmentations

Getting a vec4 using texture2d from a 2-channel texture (as in the case of 16-bit segmentations) will return a vector like (a,a,a,b), which is why the binary search for mappings on the shader failed (as an exact match of (a,b,0,0) would be needed). I solved this by allowing a new packingDegree of 2 for 16-bit layers and by transforming the volume_color before using the mapping binary search (according to the packing degree).
Therefore, 8-bit segmentations which have a packingDegree of 4 should now work as well.

### URL of deployed dev instance (used for testing):
- https://fixxbitmappings.webknossos.xyz

### Steps to test:
- Open a tracing with a 16-bit or 8-bit segmentation layer and apply a mapping:
  - Either by using merger mode in a skeleton/hybrid tracing,
  - or by using the api, like `webknossos.apiReady(3).then(async (api) => {api.data.setMapping(api.data.getVolumeTracingLayerName(), {1: 2, 2: 3}, {hideUnmappedIds: true})})`
- Mappings should work the same as for 32-bit segmentations

### Issues:
- fixes #2738 
- fixes #3636 

------
